File: app/Models/Expense.php
<?php

namespace App\Models;

use App\Enums\ExpenseStatus;
use App\Enums\ExpenseType;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Support\Facades\Storage;

class Expense extends Model
{
    protected $fillable = [
        'company_id',
        'user_id',
        'type',
        'amount',
        'currency',
        'expense_date',
        'description',
        'receipt_path',
        'receipt_original_name',
        'status',
        'reviewed_by',
        'reviewed_at',
        'manager_notes',
    ];

    protected function casts(): array
    {
        return [
            'type' => ExpenseType::class,
            'status' => ExpenseStatus::class,
            'amount' => 'decimal:2',
            'expense_date' => 'date',
            'reviewed_at' => 'datetime',
        ];
    }

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function reviewer(): BelongsTo
    {
        return $this->belongsTo(User::class, 'reviewed_by');
    }

    public function isPending(): bool
    {
        return $this->status === ExpenseStatus::Pending;
    }

    public function formattedAmount(): string
    {
        return \App\Support\Currency::format($this->amount, $this->currency);
    }

    public function receiptUrl(): ?string
    {
        if (! $this->receipt_path) {
            return null;
        }

        return Storage::disk('public')->url($this->receipt_path);
    }

    public function isImageReceipt(): bool
    {
        if (! $this->receipt_original_name) {
            return false;
        }

        $ext = strtolower(pathinfo($this->receipt_original_name, PATHINFO_EXTENSION));

        return in_array($ext, ['jpg', 'jpeg', 'png', 'gif', 'webp'], true);
    }
}

File: mobile_app/lib/models/expense.dart
class Expense {
  Expense({
    required this.id,
    required this.type,
    required this.amount,
    required this.currency,
    required this.status,
    required this.expenseDate,
    this.description,
  });

  final int id;
  final String type;
  final double amount;
  final String currency;
  final String status;
  final String expenseDate;
  final String? description;

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'] as int,
      type: json['type'] as String? ?? 'fuel',
      amount: (json['amount'] as num).toDouble(),
      currency: json['currency'] as String? ?? 'INR',
      status: json['status'] as String? ?? 'pending',
      expenseDate: json['expense_date'] as String? ?? '',
      description: json['description'] as String?,
    );
  }
}
